### PR TITLE
dev/wwolff/optimize_mram_write

### PR DESCRIPTION
**Pull request by wwolff42**
Writing to MRAM is quite tedious in the sense that address and size to MRAM must be aligned to 8 bytes.

Currently in LLDB, when writing, if address or size doesn't align to 8 bytes, we are reading the entire destination space with the constraint, and use some sort of masked data modification to finally rewrite the entire buffer back to the MRAM.

For example, taking `seqreader_example` regression test:

```
30: (lldb) memory write -i sample.bin `&buffer[0]`
30: 
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:642:WriteMemory(134217728, 0x55dc9e193400, 65501, )
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(0, 0x55dc9e193400, 65501)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 0, size rem = 5
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:642:WriteMRAM slow path final_offset = 0, padding = 0, final_size = 65504
30: 
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:642:WriteMemory(134283229, 0x55dc9e144160, 39, )
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(65501, 0x55dc9e144160, 39)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 5, size rem = 7
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:642:WriteMRAM slow path final_offset = 65496, padding = 5, final_size = 48
30: 65540 bytes were written to 0x8000000
```

LLDB internally dissect the write of 65540 bytes into two write of 65501 + 39 bytes.
Both LLDB write operation are not aligned. So in this case, we read 65504 + 48 from MRAM, update the buffer, and then write this same amount of data back to MRAM.


This patch implements a different approach by advancing the beggining and cropping the end to align them to 8 bytes, to then write a big aligned block.
On the same example as above, we only read 8 + 8 + 8 bytes for the alignement.

```
30: (lldb) memory write -i sample.bin `&buffer[0]`
30: 
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:642:WriteMemory(134217728, 0x56005774b3e0, 65501, )
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:660:WriteMemory locationInMRAM = 0
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:680:WriteMemory cropping sizeToAlignSize = 5
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(65496, 0x56005775b3b8, 5)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 0, size rem = 5
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:642:WriteMRAM slow path final_offset = 65496, padding = 0, final_size = 8
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:688:WriteMemory new size = 65496
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:691:WriteMemory big aligned block locationInMRAM = 0, size = 65496
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(0, 0x56005774b3e0, 65496)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 0, size rem = 0
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:631:WriteMRAM return fast path
30: 
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:642:WriteMemory(134283229, 0x5600576fc160, 39, )
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:660:WriteMemory locationInMRAM = 65501
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:667:WriteMemory advancing offsetToAlignLocation = 3
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(65501, 0x5600576fc160, 3)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 5, size rem = 3
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:642:WriteMRAM slow path final_offset = 65496, padding = 5, final_size = 8
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:673:WriteMemory new locationInMRAM = 65504
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:674:WriteMemory new size = 36
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:680:WriteMemory cropping sizeToAlignSize = 4
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(65536, 0x5600576fc183, 4)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 0, size rem = 4
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:642:WriteMRAM slow path final_offset = 65536, padding = 0, final_size = 8
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:688:WriteMemory new size = 32
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:691:WriteMemory big aligned block locationInMRAM = 65504, size = 32
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:621:WriteMRAM(65504, 0x5600576fc163, 32)
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:622:WriteMRAM offset rem = 0, size rem = 0
30: ../../../../../work/dpu_tools/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:631:WriteMRAM return fast path
30: 65540 bytes were written to 0x8000000
```


Here is another example, exhibing the similar change on WRAM writing:
```
(lldb) memory write -i sample.bin `&buffer[2]`

../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:647:WriteMemory(450, 0x55d2f338df60, 19, )
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:448:WriteWRAMTreatingAlignement(450, 0x55d2f338df60, 19)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:449:WriteWRAMTreatingAlignement offset rem = 2, size rem = 3
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:475:WriteWRAMTreatingAlignement slow path final_offset_in_dpuword = 112, padding = 2, final_size_in_dpuword = 6
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:502:WriteWRAMTreatingAlignement return res = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:678:WriteMemory bytes_written 19
19 bytes were written to 0x1c2
(lldb) memory write -i sample.bin `&buffer_mram[2]`

../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:647:WriteMemory(134217730, 0x55d2f338df60, 19, )
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:673:WriteMRAMTreatingAlignement(2, 0x55d2f338df60, 19)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:674:WriteMRAMTreatingAlignement offset rem = 2, size rem = 3
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:694:WriteMRAMTreatingAlignement slow path final_offset = 0, padding = 2, final_size = 24
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:716:WriteMRAMTreatingAlignement return res = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:678:WriteMemory bytes_written 19
19 bytes were written to 0x8000002
```
vs

```
(lldb) memory write -i sample.bin `&buffer[2]`                                                                                                                                                                      
                                                                                                                                                                                                                    
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:647:WriteMemory(450, 0x562d26032f60, 19, )
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:508:WriteWRAM locationInWRAM = 450                     
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:515:WriteWRAM advancing offsetToAlignLocation = 2
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:448:WriteWRAMTreatingAlignement(450, 0x562d26032f60, 2)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:449:WriteWRAMTreatingAlignement offset rem = 2, size rem = 2
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:475:WriteWRAMTreatingAlignement slow path final_offset_in_dpuword = 112, padding = 2, final_size_in_dpuword = 1
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:502:WriteWRAMTreatingAlignement return res = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:521:WriteWRAM new locationInWRAM = 452
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:522:WriteWRAM new size = 17
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:528:WriteWRAM cropping sizeToAlignSize = 1
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:448:WriteWRAMTreatingAlignement(468, 0x562d26032f72, 1)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:449:WriteWRAMTreatingAlignement offset rem = 0, size rem = 1
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:475:WriteWRAMTreatingAlignement slow path final_offset_in_dpuword = 117, padding = 0, final_size_in_dpuword = 1
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:502:WriteWRAMTreatingAlignement return res = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:536:WriteWRAM new size = 16
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:539:WriteWRAM big aligned block locationInWRAM = 452, size = 16
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:448:WriteWRAMTreatingAlignement(452, 0x562d26032f62, 16)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:449:WriteWRAMTreatingAlignement offset rem = 0, size rem = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:460:WriteWRAMTreatingAlignement return fast path
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:678:WriteMemory bytes_written 19
19 bytes were written to 0x1c2
(lldb) memory write -i sample.bin `&buffer_mram[2]`

../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:647:WriteMemory(134217730, 0x562d26032f60, 19, )
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:721:WriteMRAM locationInMRAM = 2
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:727:WriteMRAM advancing offsetToAlignLocation = 6
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:672:WriteMRAMTreatingAlignement(2, 0x562d26032f60, 6)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:673:WriteMRAMTreatingAlignement offset rem = 2, size rem = 6
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:693:WriteMRAMTreatingAlignement slow path final_offset = 0, padding = 2, final_size = 8
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:715:WriteMRAMTreatingAlignement return res = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:733:WriteMRAM new locationInMRAM = 8
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:734:WriteMRAM new size = 13
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:740:WriteMRAM cropping sizeToAlignSize = 5
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:672:WriteMRAMTreatingAlignement(16, 0x562d26032f6e, 5)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:673:WriteMRAMTreatingAlignement offset rem = 0, size rem = 5
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:693:WriteMRAMTreatingAlignement slow path final_offset = 16, padding = 0, final_size = 8
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:715:WriteMRAMTreatingAlignement return res = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:748:WriteMRAM new size = 8
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:751:WriteMRAM big aligned block locationInMRAM = 8, size = 8
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:672:WriteMRAMTreatingAlignement(8, 0x562d26032f66, 8)
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:673:WriteMRAMTreatingAlignement offset rem = 0, size rem = 0
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/Dpu.cpp:682:WriteMRAMTreatingAlignement return fast path
../../../../../work/dpu_tools_optimize_mram_write/llvm-project/lldb/source/Plugins/Process/Dpu/ProcessDpu.cpp:678:WriteMemory bytes_written 19
19 bytes were written to 0x8000002
```

### CODE REVIEW

**Comment by wwolff42** [comment]
The commented `printf` and conditional will be removed before merge.
I keep them for review and discussion for now.

**Comment by wwolff42** [comment]
We could apply this approach to both WriteIRAM and WriteWRAM.

I will rework that to keep it simple in ProcessDPU, and add wrapper function on top of WriteMRAM, WriteIRAM and WriteWRAM.

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 1]
```diff
@@ -487,9 +499,49 @@ bool Dpu::WriteWRAM(uint32_t offset, const void *buf, size_t size) {
                                  final_size_in_dpuword);
 
   delete[] words;
+  // printf("%s:%d:%s return res = %d\n", __FILE__, __LINE__, __func__, ret);
   return ret == DPU_OK;
 }
 
+bool Dpu::WriteWRAM(uint32_t offset, const void *buf, size_t size) {
+  return this->WriteWRAMTreatingAlignement(offset, buf, size);
```
Comment (SylvanBrocard) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp]: I assume everything after that is leftover debug? Or is this early return the debug state?

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 2]
```diff
@@ -444,7 +444,11 @@ StateType Dpu::StepThread(uint32_t thread_index, unsigned int *exit_status) {
   return StateType::eStateStopped;
 }
 
-bool Dpu::WriteWRAM(uint32_t offset, const void *buf, size_t size) {
+bool Dpu::WriteWRAMTreatingAlignement(uint32_t offset, const void *buf, size_t size) {
+  // printf("%s:%d:%s(%u, %p, %zu)\n", __FILE__, __LINE__, __func__, offset, buf, size);
```
Comment (SylvanBrocard) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp]: Let's do another review after the debug prints cleanup.

File: lldb/source/Plugins/Process/Dpu/Dpu.h [hunk 3]
```diff
@@ -74,10 +74,12 @@ class Dpu {
                            const uint32_t current_pc);
   lldb::StateType StepThread(uint32_t thread_index, unsigned int *exit_status);
 
+  bool WriteWRAMTreatingAlignement(uint32_t offset, const void *buf, size_t size);
```
Comment (SylvanBrocard) [inline lldb/source/Plugins/Process/Dpu/Dpu.h]: Alignement -> Alignment

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 4]
```diff
@@ -487,9 +499,49 @@ bool Dpu::WriteWRAM(uint32_t offset, const void *buf, size_t size) {
...
+    pointerOffsetBuf += offsetToAlignLocation;
+    // printf("%s:%d:%s new locationInWRAM = %u\n", __FILE__, __LINE__, __func__, locationInWRAM);
+    // printf("%s:%d:%s new size = %zu\n", __FILE__, __LINE__, __func__, size);
+  }
+
+  // align size
+  size_t sizeToAlignSize = size % sizeof(dpuword_t);
+  if (sizeToAlignSize != 0) {
+    // printf("%s:%d:%s cropping sizeToAlignSize = %zu\n", __FILE__, __LINE__, __func__, sizeToAlignSize);
+    const void *localBufPointer = (const char*)buf + pointerOffsetBuf + (size - sizeToAlignSize);
```
Comment (SylvanBrocard) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp]: Use static_cast<const char*> over C-style cast
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-cstyle-cast.html

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 5]
```diff
@@ -487,9 +499,49 @@ bool Dpu::WriteWRAM(uint32_t offset, const void *buf, size_t size) {
...
+  uint32_t locationInWRAM = offset;
+  // printf("%s:%d:%s locationInWRAM = %u\n", __FILE__, __LINE__, __func__, locationInWRAM);
+
+  size_t pointerOffsetBuf = 0;
+  size_t offsetToAlignLocation = sizeof(dpuword_t) - (locationInWRAM % sizeof(dpuword_t));
+
+  // align addr
+  if ((offsetToAlignLocation != sizeof(dpuword_t)) && (offsetToAlignLocation < size)) {
+    // printf("%s:%d:%s advancing offsetToAlignLocation = %zu\n", __FILE__, __LINE__, __func__, offsetToAlignLocation);
+    if (!this->WriteWRAMTreatingAlignement(locationInWRAM, buf, offsetToAlignLocation))
```
Comment (SylvanBrocard) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp]: `this->` is unnecessary inside methods definition

**Review by SylvanBrocard — CHANGES_REQUESTED** [review]

**Review by SylvanBrocard — APPROVED** [review]

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 6]
```diff
@@ -444,7 +444,7 @@ StateType Dpu::StepThread(uint32_t thread_index, unsigned int *exit_status) {
   return StateType::eStateStopped;
 }
 
-bool Dpu::WriteWRAM(uint32_t offset, const void *buf, size_t size) {
+bool Dpu::WriteWRAMTreatingAlignment(uint32_t offset, const void *buf, size_t size) {
```
Comment (placouture-upmem) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp]: I'm not sure that "Treating" is how English would put it. See https://www.linguee.com/english-french/translation/treat.html

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 7]
```diff
@@ -617,7 +649,7 @@ bool Dpu::ReadIRAM(uint32_t offset, void *buf, size_t size) {
   return ret == DPU_OK;
 }
 
-bool Dpu::WriteMRAM(uint32_t offset, const void *buf, size_t size) {
+bool Dpu::WriteMRAMTreatingAlignment(uint32_t offset, const void *buf, size_t size) {
```
Comment (placouture-upmem) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp]: Same comment as above regarding the "Treating" term.

File: lldb/source/Plugins/Process/Dpu/Dpu.cpp [hunk 8]
```diff
@@ -654,6 +686,38 @@ bool Dpu::WriteMRAM(uint32_t offset, const void *buf, size_t size) {
   return ret == DPU_OK;
 }
 
+bool Dpu::WriteMRAM(uint32_t offset, const void *buf, size_t size) {
```
Comment (placouture-upmem) [inline lldb/source/Plugins/Process/Dpu/Dpu.cpp:689]: Shouldn't we insert a doxygen-like function header comment, that explains shortly the processing difference between `Write[MW]RAM` and `Write[MW]RAMTreatingAlignment`?

Something like:

```
/**
 * @brief Write a buffer of bytes from SRAM to MRAM.
 * This function intends to use optimized write operations
 * for the part of the buffer that allows complete word-aligned read and
 * write operations
 * @param offset The offset of the first byte to be written to in MRAM
 * @buf The bytes buffer to be written from in SRAM
 * @size The number of bytes to be written
 * @return Whether the operation was succesful.
```

```
/**
 * @brief Write a buffer of bytes from SRAM to MRAM.
 * This function operates properly on buffers either word-aligned or not.
 * However, it does not attempt to fully optimize the write operations for
 * large buffers that are not properly word-aligned.
 * @param offset The offset of the first byte to be written to in MRAM
 * @buf The bytes buffer to be written from in SRAM
 * @size The number of bytes to be written
 * @return Whether the operation was succesful.
```

**Review by placouture-upmem — APPROVED** [review]
Approved, with suggestions about function naming and comments.

**Pull request merged**